### ISSUE TITLE
Docker swarm, node unable to fetch the image from the machine's local repository.

### ISSUE THREAD

**Issue by shreyas-badiger** [moby/moby#35568]
.>>>>>>>>>>>>>>>>>>>> I am trying to  create a container with the service command but the container is not able to get the specified image though it is present in the node's local repo.

docker service create --name container2 rastasheep/ubuntu-sshd:14.04 orion-01.local
782ayuxb75sdkh0c0qlloa88s
Since --detach=false was not specified, tasks will be created in the background.
In a future release, --detach=false will become the default.

docker service ps container2
yq7kyv54rt2q         \_ container2.1    rastasheep/ubuntu-sshd:14.04   orion-02.local       Shutdown            Rejected 1 second ago      "No such image: rastasheep/
ubu…"
u8xv03u3vrhb         \_ container2.1    rastasheep/ubuntu-sshd:14.04   orion-01.local	    Shutdown            Rejected 28 seconds ago      "No such image: rastasheep/
ubu…"

.>>>>>>>>>>>>>>>>>>>>But the image is already present in the node's local repo..

[root@orion-01 shreyas]# hostname
orion-01.local
[root@orion-01 shreyas]# docker images
REPOSITORY               TAG                 IMAGE ID            CREATED             SIZE
rastasheep/ubuntu-sshd   14.04               7d6785aa54d6        4 days ago          284 MB
[root@orion-02 shreyas]# hostname
orion-02.local
[root@orion-02 shreyas]# docker images
REPOSITORY               TAG                 IMAGE ID            CREATED             SIZE
rastasheep/ubuntu-sshd   14.04               7d6785aa54d6        4 days ago          284 MB

**Comment by hard-fault** [comment]
Note: Both orion-01.local and orion-02.local
nodes are not connected to the internet. I have copied the image from other node using "docker save" and "docker load"

**Comment by hard-fault** [comment]
Is it necessary to have same docker version on all the nodes in swarm?

**Comment by thaJeztah** [comment]
Please provide more information as requested in the [issue-template](https://raw.githubusercontent.com/docker/docker/master/.github/ISSUE_TEMPLATE.md) that was shown when you opened this issue.

- provide the output of `docker version`
- provide the output of `docker info`

**Comment by hard-fault** [comment]
<<<<<<<<<<<<<<<<<<<<<HEADNODE>>>>>>>>>>>>>>>>>>>>>>>>>>>>
[root@rigel Demo]# docker version
Client:
 Version:      17.09.0-ce
 API version:  1.32
 Go version:   go1.8.3
 Git commit:   afdb6d4
 Built:        Tue Sep 26 22:41:23 2017
 OS/Arch:      linux/amd64

Server:
 Version:      17.09.0-ce
 API version:  1.32 (minimum version 1.12)
 Go version:   go1.8.3
 Git commit:   afdb6d4
 Built:        Tue Sep 26 22:42:49 2017
 OS/Arch:      linux/amd64
 Experimental: false
[root@rigel Demo]# docker info
Containers: 14
 Running: 6
 Paused: 0
 Stopped: 8
Images: 15
Server Version: 17.09.0-ce
Storage Driver: devicemapper
 Pool Name: docker-8:48-5017351295-pool
 Pool Blocksize: 65.54kB
 Base Device Size: 10.74GB
 Backing Filesystem: xfs
 Data file: /dev/loop0
 Metadata file: /dev/loop1
 Data Space Used: 3.818GB
 Data Space Total: 107.4GB
 Data Space Available: 103.6GB
 Metadata Space Used: 5.235MB
 Metadata Space Total: 2.147GB
 Metadata Space Available: 2.142GB
 Thin Pool Minimum Free Space: 10.74GB
 Udev Sync Supported: true
 Deferred Removal Enabled: true
 Deferred Deletion Enabled: true
 Deferred Deleted Device Count: 6
 Data loop file: /scratch/docker/devicemapper/devicemapper/data
 Metadata loop file: /scratch/docker/devicemapper/devicemapper/metadata
 Library Version: 1.02.140-RHEL7 (2017-05-03)
Logging Driver: json-file
Cgroup Driver: cgroupfs
Plugins:
 Volume: local
 Network: bridge host macvlan null overlay
 Log: awslogs fluentd gcplogs gelf journald json-file logentries splunk syslog
Swarm: active
 NodeID: rtfy95p5s42mdrljh2t2ashp0
 Is Manager: true
 ClusterID: kttag1eehv57e8ofjpfgqencu
 Managers: 1
 Nodes: 3
 Orchestration:
  Task History Retention Limit: 5
 Raft:
  Snapshot Interval: 10000
  Number of Old Snapshots to Retain: 0
  Heartbeat Tick: 1
  Election Tick: 3
 Dispatcher:
  Heartbeat Period: 5 seconds
 CA Configuration:
  Expiry Duration: 3 months
  Force Rotate: 0
 Autolock Managers: false
 Root Rotation In Progress: false
 Node Address: 192.168.0.10
 Manager Addresses:
  192.168.0.10:2377
Runtimes: runc
Default Runtime: runc
Init Binary: docker-init
containerd version: 06b9cb35161009dcb7123345749fef02f7cea8e0
runc version: 3f2f8b84a77f73d38244dd690525642a72156c64
init version: 949e6fa
Security Options:
 seccomp
  Profile: default
Kernel Version: 3.10.0-123.20.1.el7.x86_64
Operating System: CentOS Linux 7 (Core)
OSType: linux
Architecture: x86_64
CPUs: 32
Total Memory: 125.7GiB
Name: rigel.serc.iisc.in
ID: CZWQ:ELGL:OPKJ:V3B7:NSXP:OM5T:FSOW:OP77:4OQD:UOIY:YMB5:IYN3
Docker Root Dir: /scratch/docker
Debug Mode (client): false
Debug Mode (server): false
Username: shrey67
Registry: https://index.docker.io/v1/
Experimental: false
Insecure Registries:
 127.0.0.0/8
Live Restore Enabled: false

WARNING: devicemapper: usage of loopback devices is strongly discouraged for production use.
         Use `--storage-opt dm.thinpooldev` to specify a custom block storage device.

<<<<<<<<<<<<<<<<<<<<<orion-01.local>>>>>>>>>>>>>>>>>>>>>>>>>>>>
[root@rigel Demo]# ssh node1
Last login: Wed Nov 22 14:49:37 2017 from rigel.serc.iisc.in
[root@orion-01 ~]# docker version
Client:
 Version:         1.12.6
 API version:     1.24
 Package version: docker-1.12.6-55.gitc4618fb.el7.centos.x86_64
 Go version:      go1.8.3
 Git commit:      c4618fb/1.12.6
 Built:           Thu Sep 21 22:33:52 2017
 OS/Arch:         linux/amd64

Server:
 Version:         1.12.6
 API version:     1.24
 Package version: docker-1.12.6-55.gitc4618fb.el7.centos.x86_64
 Go version:      go1.8.3
 Git commit:      c4618fb/1.12.6
 Built:           Thu Sep 21 22:33:52 2017
 OS/Arch:         linux/amd64
[root@orion-01 ~]# docker info
Containers: 12
 Running: 1
 Paused: 0
 Stopped: 11
Images: 1
Server Version: 1.12.6
Storage Driver: devicemapper
 Pool Name: docker-8:2-5506630-pool
 Pool Blocksize: 65.54 kB
 Base Device Size: 10.74 GB
 Backing Filesystem: xfs
 Data file: /dev/loop0
 Metadata file: /dev/loop1
 Data Space Used: 721 MB
 Data Space Total: 107.4 GB
 Data Space Available: 106.7 GB
 Metadata Space Used: 2.093 MB
 Metadata Space Total: 2.147 GB
 Metadata Space Available: 2.145 GB
 Thin Pool Minimum Free Space: 10.74 GB
 Udev Sync Supported: true
 Deferred Removal Enabled: true
 Deferred Deletion Enabled: true
 Deferred Deleted Device Count: 0
 Data loop file: /var/lib/docker/devicemapper/devicemapper/data
 WARNING: Usage of loopback devices is strongly discouraged for production use. Use `--storage-opt dm.thinpooldev` to specify a custom block storage device.
 Metadata loop file: /var/lib/docker/devicemapper/devicemapper/metadata
 Library Version: 1.02.140-RHEL7 (2017-05-03)
Logging Driver: journald
Cgroup Driver: systemd
Plugins:
 Volume: local
 Network: host overlay bridge null
Swarm: active
 NodeID: t7ziws0sdqp9b3adotr3gm99l
 Is Manager: false
 Node Address: 192.168.0.11
Runtimes: runc docker-runc
Default Runtime: docker-runc
Security Options: seccomp
Kernel Version: 3.10.0-123.20.1.el7.x86_64
Operating System: CentOS Linux 7 (Core)
OSType: linux
Architecture: x86_64
Number of Docker Hooks: 3
CPUs: 8
Total Memory: 31.25 GiB
Name: orion-01.local
ID: YVYY:IULT:DD3R:JEDW:SUDH:KF5V:536P:7SSH:44XC:VX3V:ZQ2H:7LJZ
Docker Root Dir: /var/lib/docker
Debug Mode (client): false
Debug Mode (server): false
Registry: https://index.docker.io/v1/
Insecure Registries:
 127.0.0.0/8
Registries: docker.io (secure)
[root@orion-01 ~]# exit
logout
Connection to node1 closed.

<<<<<<<<<<<<<<<<<<<<<orion-02.local>>>>>>>>>>>>>>>>>>>>>>>>>>>>

[root@rigel Demo]# ssh node2
Last login: Wed Nov 22 15:06:27 2017 from rigel.serc.iisc.in
[root@orion-02 ~]# docker version
Client:
 Version:         1.12.6
 API version:     1.24
 Package version: docker-1.12.6-55.gitc4618fb.el7.centos.x86_64
 Go version:      go1.8.3
 Git commit:      c4618fb/1.12.6
 Built:           Thu Sep 21 22:33:52 2017
 OS/Arch:         linux/amd64

Server:
 Version:         1.12.6
 API version:     1.24
 Package version: docker-1.12.6-55.gitc4618fb.el7.centos.x86_64
 Go version:      go1.8.3
 Git commit:      c4618fb/1.12.6
 Built:           Thu Sep 21 22:33:52 2017
 OS/Arch:         linux/amd64
[root@orion-02 ~]# docker info
Containers: 6
 Running: 1
 Paused: 0
 Stopped: 5
Images: 1
Server Version: 1.12.6
Storage Driver: devicemapper
 Pool Name: docker-8:2-655828-pool
 Pool Blocksize: 65.54 kB
 Base Device Size: 10.74 GB
 Backing Filesystem: xfs
 Data file: /dev/loop0
 Metadata file: /dev/loop1
 Data Space Used: 1.107 GB
 Data Space Total: 107.4 GB
 Data Space Available: 105.1 GB
 Metadata Space Used: 2.077 MB
 Metadata Space Total: 2.147 GB
 Metadata Space Available: 2.145 GB
 Thin Pool Minimum Free Space: 10.74 GB
 Udev Sync Supported: true
 Deferred Removal Enabled: true
 Deferred Deletion Enabled: true
 Deferred Deleted Device Count: 0
 Data loop file: /var/lib/docker/devicemapper/devicemapper/data
 WARNING: Usage of loopback devices is strongly discouraged for production use. Use `--storage-opt dm.thinpooldev` to specify a custom block storage device.
 Metadata loop file: /var/lib/docker/devicemapper/devicemapper/metadata
 Library Version: 1.02.140-RHEL7 (2017-05-03)
Logging Driver: journald
Cgroup Driver: systemd
Plugins:
 Volume: local
 Network: overlay host bridge null
Swarm: active
 NodeID: whnx12kvb5gd6xvdf6rb00j3k
 Is Manager: false
 Node Address: 192.168.0.12
Runtimes: docker-runc runc
Default Runtime: docker-runc
Security Options: seccomp
Kernel Version: 3.10.0-123.20.1.el7.x86_64
Operating System: CentOS Linux 7 (Core)
OSType: linux
Architecture: x86_64
Number of Docker Hooks: 3
CPUs: 8
Total Memory: 31.25 GiB
Name: orion-02.local
ID: NLEQ:ONRX:4MRJ:23CI:AQ5A:B7F6:LQ4T:CK2F:O7JO:7TST:P4NP:52AU
Docker Root Dir: /var/lib/docker
Debug Mode (client): false
Debug Mode (server): false
Registry: https://index.docker.io/v1/
WARNING: bridge-nf-call-ip6tables is disabled
Insecure Registries:
 127.0.0.0/8
Registries: docker.io (secure)

**Comment by zmirc** [comment]
A similar thing happened to me today with 17.09.0-ce:
I started a swarm service constrained to go on only 3 worker servers (based on docker engine labels). On 2 of them it started properly, but on the 3rd one it failed to download from the private repository (which also runs as a service in the same cluster).
It worked after restarting docker on the 3rd worker.

Then after restarting a few more servers, the custom network I had created in the swarm disappeared on one of the worker nodes. (I know it might be a different issue, but good to mention nevertheless) (I don't have a reproducible scenario yet)
